### PR TITLE
Fix `calculate_age_at_registration`

### PR DESCRIPTION
**Pull request by pavlo-mk**
Fix `calculate_age_at_registration`

Usually we are using `calculate_age_at_registration ` to calculate `age_at_registration` and field is `age_at_registration = models.PositiveSmallIntegerField(null=True, blank=True, help_text="Age at registration [sys]")` so we can have any values below 0

[AB#309211](https://unicef.visualstudio.com/ICTD-HCT-MIS/_workitems/edit/309211): Angola: Kobo issue "Import Error" - example was this issue with kobo import

### CODE REVIEW

File: tests/unit/apps/utils/test_age_at_registration.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from datetime import datetime
+
+from hope.apps.utils.age_at_registration import calculate_age_at_registration
+
+
+def test_calculate_age_at_registration() -> None:
```
Comment (johniak) [inline tests/unit/apps/utils/test_age_at_registration.py]: Per project test conventions, one test should equal one scenario (Arrange-Act-Assert). This single test function covers three distinct scenarios (ValueError, future date, happy path). These should be split into three separate test functions or use pytest.mark.parametrize:

File: tests/unit/apps/utils/test_age_at_registration.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from datetime import datetime
```
Comment (johniak) [inline tests/unit/apps/utils/test_age_at_registration.py:1]: Consider adding a test case for birth_date == created_at (same day), which should return 0. This is a meaningful edge case for the boundary check on L15 of the source (birth_date_parsed > created_at_date).

**Review by johniak — CHANGES_REQUESTED** [review]

File: tests/unit/apps/utils/test_age_at_registration.py (hunk 1 above)
Comment (domdinicola) [reply tests/unit/apps/utils/test_age_at_registration.py]: +1

File: tests/unit/apps/utils/test_age_at_registration.py (hunk 2 above)
Comment (domdinicola) [reply tests/unit/apps/utils/test_age_at_registration.py:1]: agree

**Review by domdinicola — CHANGES_REQUESTED** [review]

**Review by johniak — APPROVED** [review]

**Comment by domdinicola** [comment]
@saxix does this make sense to you?

**Comment by domdinicola** [comment]
@pavlo-mk should not raise a validation error?

**Comment by pavlo-mk** [comment]
it's better to have ValueError because of the `parser.parse(birth_date)` in my opinion, ValidationError can be added if result of calculate_age_at_registration is less then 0
